### PR TITLE
feat(TMRX-1390): scrollable l2 nav

### PR DESCRIPTION
**Pull request by yukiinasawa**
### Description

scrollable l2 nav

[TMRX-1390](https://nidigitalsolutions.jira.com/browse/TMRX-1390)

### Checklist

- [x] Have you done any manual testing?
- [ ] Does it have automated tests?
- [ ] Do you need any other PRs merged before this (if so please list)?
- [ ] Do you need to update the README/Runbook
- [ ] Have you checked for [accessibility](https://nidigitalsolutions.jira.com/wiki/spaces/TNLDigital/pages/3898048523/Accessibility+Dev+Guide#Checklist)?

### Screenshots (if appropriate):

Include screenshots if needed.

[TMRX-1390]: https://nidigitalsolutions.jira.com/browse/TMRX-1390?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/ts-newskit/src/components/navigation/secondary-menu/index.tsx [hunk 1]
```diff
@@ -10,13 +10,17 @@ interface SecondaryNavigationProps {
   pageSlug: string;
   stickyTop?: number;
   stickyTopDesktop?: number;
+  onClick?: (isExpanded: boolean) => void
+  withScroll?: boolean
```
Comment (ibrahim-kurhan) [inline packages/ts-newskit/src/components/navigation/secondary-menu/index.tsx:14]: so `withScroll` is optional right?

File: packages/ts-newskit/src/components/navigation/secondary-menu/styles.ts [hunk 2]
```diff
@@ -95,3 +96,8 @@ export const NavItemsContainer = styled.div`
   z-index: 1;
   ${getColorCssFromTheme('backgroundColor', 'white')};
 `;
+
+export const NavItemMobileContainer = styled.div<NavItemMobileContainerProp>`
+    height:  ${({ height }) => height};
+    overflow-y: ${({ withScroll }) => `${withScroll ? 'scroll': 'visible'}`};
```
Comment (ibrahim-kurhan) [inline packages/ts-newskit/src/components/navigation/secondary-menu/styles.ts:102]: do you need template literal? could this be `overflow-y: ${({ withScroll }) => (withScroll ? 'scroll' : 'visible')};`

File: packages/ts-newskit/src/components/navigation/secondary-menu/mobile/index.tsx [hunk 3]
```diff
@@ -1,30 +1,60 @@
...
 export const SecondaryNavMobile: React.FC<{
-  options: SecondaryMenuOptions;
-  data: SecondaryMenuItem[];
-}> = ({ options, data }) => {
-  const { isExpanded, isSelected } = options;
-  const subMenuTitle = isExpanded ? 'Close' : 'See all';
+    options: SecondaryMenuOptions;
+    data: SecondaryMenuItem[];
+    onClick?: (isExpanded: boolean) => void;
+    withScroll: boolean;
```
[the remark this answers is not in the document]
Comment (yukiinasawa) [reply packages/ts-newskit/src/components/navigation/secondary-menu/mobile/index.tsx:12]: Since we set default to `false` in parent component, it always has value. But I'm happy to make it optional if we want :)

File: packages/ts-newskit/src/components/navigation/secondary-menu/mobile/navigator.tsx [hunk 4]
```diff
@@ -27,6 +28,7 @@ export const Navigator: React.FC<{
         expanded={isExpanded}
         onClick={() => {
           setIsExpanded(!isExpanded);
+          onClick?.(!isExpanded)
```
Comment (ibrahim-kurhan) [inline packages/ts-newskit/src/components/navigation/secondary-menu/mobile/navigator.tsx:31]: is linting happy with optional chaining?

Comment (yukiinasawa) [reply packages/ts-newskit/src/components/navigation/secondary-menu/mobile/navigator.tsx:31]: Nope 😭

File: packages/ts-newskit/src/components/navigation/secondary-menu/desktop/navItems.tsx [hunk 5]
```diff
@@ -21,7 +21,7 @@ export const NavItems: React.FC<{
           }}
           href={url}
           key={url}
-          onClick={() => handleSelect(slug)}
+          onClick={() => handleSelect(title)}
```
Comment (douglasmik) [inline packages/ts-newskit/src/components/navigation/secondary-menu/desktop/navItems.tsx:24]: was this change required?

File: packages/ts-newskit/src/components/navigation/secondary-menu/index.tsx (hunk 1 above)
Comment (douglasmik) [reply packages/ts-newskit/src/components/navigation/secondary-menu/index.tsx:14]: what does this do - should this not always be scrollable?

File: packages/ts-newskit/src/components/navigation/secondary-menu/mobile/index.tsx [hunk 6]
```diff
@@ -1,30 +1,60 @@
...
+            ) {
+                setHeight(
+                    `${
+                        window.innerHeight -
+                        navRef.current.getBoundingClientRect().top
+                    }px`
+                );
+            } else setHeight('100vh');
+        }
+    }, [navRef?.current]);
```
Comment (douglasmik) [inline packages/ts-newskit/src/components/navigation/secondary-menu/mobile/index.tsx:33]: might be better to set height based on the number of items;
e.g. `data.length x 48`px

File: packages/ts-newskit/src/components/navigation/secondary-menu/desktop/navItems.tsx (hunk 5 above)
Comment (yukiinasawa) [reply packages/ts-newskit/src/components/navigation/secondary-menu/desktop/navItems.tsx:24]: This is the change we discussed the other day to fix highlight issue by puzzles team.

File: packages/ts-newskit/src/components/navigation/secondary-menu/index.tsx (hunk 1 above)
Comment (yukiinasawa) [reply packages/ts-newskit/src/components/navigation/secondary-menu/index.tsx:14]: AFAIK not always - as this components is been used for both mobile and desktop.

File: packages/ts-newskit/src/components/navigation/secondary-menu/styles.ts (hunk 2 above)
Comment (douglasmik) [reply packages/ts-newskit/src/components/navigation/secondary-menu/styles.ts:102]: do wee need this as visible/scroll? Linked to question above - do we need `withScroll` prop?

File: packages/ts-newskit/src/components/navigation/secondary-menu/index.tsx (hunk 1 above)
Comment (yukiinasawa) [reply packages/ts-newskit/src/components/navigation/secondary-menu/index.tsx:14]: ^ actually is for only mobile, I thought it would be nice to make it flexible - so could be non-scrollable. But happy to be always scrollable if we want it to be.

File: packages/ts-newskit/src/components/navigation/secondary-menu/styles.ts (hunk 2 above)
Comment (douglasmik) [reply packages/ts-newskit/src/components/navigation/secondary-menu/styles.ts:102]: f set to `auto` it should only apply scroll if needed and will be `visible` if content is not clipped. That way, we could remove the `withScroll` prop

File: packages/ts-newskit/src/components/navigation/secondary-menu/desktop/navItems.tsx (hunk 5 above)
Comment (douglasmik) [reply packages/ts-newskit/src/components/navigation/secondary-menu/desktop/navItems.tsx:24]: I thought them changing from `title` to `slug` was part of the fix? Might need to check with them?

File: packages/ts-newskit/src/components/navigation/secondary-menu/mobile/index.tsx (hunk 6 above)
Comment (douglasmik) [reply packages/ts-newskit/src/components/navigation/secondary-menu/mobile/index.tsx:33]: correct, understand what this is doing now. Was getting the menu jumping to the bottom, which is more the issue, as `NavItemsContainer` is set to 100%. As this is now in a sub container, top should now be the explicit height of the sub nav bar (48px)

File: packages/ts-newskit/src/components/navigation/secondary-menu/desktop/navItems.tsx (hunk 5 above)
Comment (douglasmik) [reply packages/ts-newskit/src/components/navigation/secondary-menu/desktop/navItems.tsx:24]: Think we need to update mobile/navitems to `slug` also

File: packages/ts-newskit/src/components/navigation/secondary-menu/styles.ts [hunk 7]
```diff
@@ -95,3 +96,8 @@ export const NavItemsContainer = styled.div`
   z-index: 1;
   ${getColorCssFromTheme('backgroundColor', 'white')};
 `;
+
+export const NavItemMobileContainer = styled.div<NavItemMobileContainerProp>`
+    height:  ${({ height }) => height};
+    overflow-y: 'scroll';
```
Comment (douglasmik) [inline packages/ts-newskit/src/components/navigation/secondary-menu/styles.ts:102]: would put this to `auto` to accommodate both states where it needs to scroll and where items are all visible within the space

File: packages/ts-newskit/src/components/navigation/secondary-menu/types.ts [hunk 8]
```diff
@@ -27,3 +27,7 @@ export type SecondaryMenuOptions = {
   isExpanded: boolean;
   setIsExpanded: (value: boolean) => void;
 };
+
+export type NavItemMobileContainerProp = {
+  $height: string;
```
[the remark this answers is not in the document]
Comment (yukiinasawa) [reply packages/ts-newskit/src/components/navigation/secondary-menu/types.ts:32]: https://styled-components.com/docs/api#transient-props

**Review by ibrahim-kurhan — APPROVED** [review]

**Review by douglasmik — APPROVED** [review]
